backend/core/tests/test_contributors.py

```diff
@@ -0,0 +1,239 @@
+    # For these tests, we'll use the Episode model.
+
+    UPDATE_EPISODE_MUTATION = """
+        mutation UPDATE_EPISODE($input: UpdateEpisodeMutationInput!) {
+            updateEpisode(input: $input) {
+                ok
+                errors {
+                    field
+                    messages
+                }
+            }
+        }
+    """
+
+    @classmethod
+    def setUpTestData(cls) -> None:
+        source = Source.objects.create(name="A New Hope")
+        cls.episode = Episode.objects.create(
+            name="Death Star destroyed", source_id=source.pk
+        )
+        cls.user_1 = User.objects.create(username="Yoda")
+        cls.user_2 = User.objects.create(username="Obi-Wan")
+
+    def test_single_contributor_one_contribution(self):
+        self.client.force_login(self.user_1)
+
+        response = self.query(
+            query=self.UPDATE_EPISODE_MUTATION,
+            input_data={"id": self.episode.pk, "name": "Palpatine"},
+        )
+
+        self.assertResponseNoErrors(response)
+
+        content = json.loads(response.content)
+
+        self.assertTrue(content["data"]["updateEpisode"]["ok"])
+        self.assertEqual(len(content["data"]["updateEpisode"]["errors"]), 0)
+
+        episode = Episode.objects.get(id=self.episode.pk)
+
+        self.assertEqual(episode.name, "Palpatine")
+
+        first_contributor = episode.contributors.first()
+
+        if not first_contributor:
+            self.fail("No contributors found after mutation.")
+
+        self.assertEqual(first_contributor.username, "Yoda")
+
+    def test_one_contributor_multiple_contributions(self):
+        self.client.force_login(self.user_1)
+
+        response = self.query(
+            query=self.UPDATE_EPISODE_MUTATION,
+            input_data={"id": self.episode.pk, "name": "Palpatine"},
+        )
+
+        self.assertResponseNoErrors(response)
+
+        content = json.loads(response.content)
+
+        self.assertTrue(content["data"]["updateEpisode"]["ok"])
+        self.assertEqual(len(content["data"]["updateEpisode"]["errors"]), 0)
+
+        episode = Episode.objects.get(id=self.episode.pk)
+
+        self.assertEqual(episode.name, "Palpatine")
+        self.assertEqual(len(episode.contributors.all()), 1)
+
+        response = self.query(
+            query=self.UPDATE_EPISODE_MUTATION,
+            input_data={"id": self.episode.pk, "name": "Darth Vader"},
+        )
+
+        self.assertResponseNoErrors(response)
+
+        content = json.loads(response.content)
+
+        self.assertTrue(content["data"]["updateEpisode"]["ok"])
+        self.assertEqual(len(content["data"]["updateEpisode"]["errors"]), 0)
+
+        episode = Episode.objects.get(id=self.episode.pk)
+
+        self.assertEqual(episode.name, "Darth Vader")
+        self.assertEqual(
+            len(episode.contributors.all()),
+            1,
+            "Contributors should not increase after multiple contributions.",
+        )
```

Comment:
Since `episode.contributors` is a many-to-many field, you can't get duplicates. To be honest, I don't find it necessary to test this, since we can trust that Django (or PostgreSQL) has taken care of that; it's not something we're handling in our own code.

That said,  there is little harm in superfluous testing, so this isn't a major issue.